pkg/watch/errorlist.go

```diff
@@ -15,22 +15,91 @@ limitations under the License.
 
 package watch
 
-import "strings"
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
 
 // errorList is an error that aggregates multiple errors.
-type errorList []error
+type errorList struct {
```

Comment:
Why not make this type public so it can be used in tests?  Its internal data is protected.